packages/api/src/command/medical/facility/update-facility.ts

```diff
@@ -1,65 +1,41 @@
-import { Facility } from "../../../domain/medical/facility";
+import { FacilityCreate } from "../../../domain/medical/facility";
 import { FacilityModel } from "../../../models/medical/facility";
 import { validateVersionForUpdate } from "../../../models/_default";
 import { BaseUpdateCmdWithCustomer } from "../base-update-command";
-import { validateCreate } from "./create-facility";
+import { validateObo, validateNPI } from "./create-facility";
 import { getFacilityOrFail } from "./get-facility";
 
-type PartialProps =
-  | "eTag"
-  | "cqType"
-  | "cwType"
-  | "cqActive"
-  | "cwActive"
-  | "cqOboOid"
-  | "cwOboOid";
-
 export type FacilityUpdateCmd = BaseUpdateCmdWithCustomer &
-  Omit<Facility, "oid" | "facilityNumber" | "createdAt" | "updatedAt" | PartialProps> &
-  Partial<Pick<Facility, PartialProps>>;
-
-export const updateFacility = async (facilityUpdate: FacilityUpdateCmd): Promise<FacilityModel> => {
-  const { id, cxId, eTag } = facilityUpdate;
-
+  Partial<Omit<FacilityCreate, "cwType" | "cqType">>;
+
+export async function updateFacility({
+  id,
+  eTag,
+  cxId,
+  data,
+  cqApproved,
+  cqActive,
+  cqOboOid,
+  cwApproved,
+  cwActive,
+  cwOboOid,
+}: FacilityUpdateCmd): Promise<FacilityModel> {
   const facility = await getFacilityOrFail({ id, cxId });
   validateVersionForUpdate(facility, eTag);
-
-  const { data, cqActive, cwActive, cqOboOid, cwOboOid } = validateUpdate(facility, facilityUpdate);
-  const { name, npi, tin, active, address } = data;
-
-  return facility.update({
-    data: {
-      name,
-      npi,
-      tin,
-      active,
-      address,
-    },
-    cqActive,
-    cwActive,
+  validateObo({
```

Comment:
Given this is a generic updateFacility why would we always validate obo when it can be something else?

Reply:
Not sure I totally follow -- we're validating that the update to the facility passes the Obo check..

Do you want a to check the resulting type first before calling this? Not sure it's necessary as the validation function will just return true if the type is non-obo (which is what we want).

Reply:
What i mean is what happens if we update a facility thats not an obo

Reply:
it returns true...